test/Collector.test.ts

```diff
@@ -0,0 +1,101 @@
+import { expect, use } from 'chai';
+import Collector from '../build/Collector.json';
+import BurnableTestToken from '../build/BurnableTestToken.json';
+import { deployContract, MockProvider, solidity } from 'ethereum-waffle';
+
+use(solidity);
+
+describe('Collector', function () {
+    async function deployCollector(
+        share1?: number,
+        share2?: number,
+        share3?: number,
+        share4?: number
+    ) {
+        const [owner, partner1, partner2, partner3, partner4] =
+            new MockProvider().getWallets();
+        const partners = [
+            {
+                beneficiary: await partner1.getAddress(),
+                share: share1 ?? 20
+            },
+            {
+                beneficiary: await partner2.getAddress(),
+                share: share2 ?? 35
+            },
+            {
+                beneficiary: await partner3.getAddress(),
+                share: share3 ?? 13
+            },
+            {
+                beneficiary: await partner4.getAddress(),
+                share: share4 ?? 32
+            }
+        ];
+
+        const testToken = await deployContract(owner, BurnableTestToken);
+        const collector = await deployContract(owner, Collector, [
+            owner.address,
+            testToken.address,
+            partners
+        ]);
+
+        return { collector, testToken, partners, owner };
+    }
+
+    describe('Deployment', function () {
+        it('Should deploy with owner, token and revenue partners', async function () {
+            const { collector } = await deployCollector();
+            expect(collector.address).to.be.properAddress;
+        });
+
+        it('Should not let deploy with invalid shares', async function () {
```

Comment:
Thank you for adding these tests. What do you think if we cover more scenarios? e.g.: negative shares, no shares, etc...